pkg/shell/shell.go

```diff
@@ -22,6 +22,9 @@ type ShellConfig struct {
 
 func RunShell(config ShellConfig) error {
 	db, err := db.NewDb(config.DbPath)
+	if err := db.TestConnection(); err != nil {
+		db = nil
+	}
 	if err != nil {
 		return err
 	}
```

Comment:
We should check db.NewDb first
```suggestion
	db, err := db.NewDb(config.DbPath)
	if err != nil {
		return err
	}
	if err := db.TestConnection(); err != nil {
		db = nil
	}
```

Reply:
True! Thx for pointing this out. Fixed!